scripts/lib/CIME/case_st_archive.py

```diff
@@ -7,15 +7,43 @@
 from CIME.XML.standard_module_setup import *
 from CIME.case_submit               import submit
 from CIME.XML.env_archive           import EnvArchive
-from CIME.utils                     import run_and_log_case_status, ls_sorted_by_mtime
+from CIME.utils                     import run_and_log_case_status, ls_sorted_by_mtime, symlink_force
 from os.path                        import isdir, join
 import datetime
 
 logger = logging.getLogger(__name__)
 
 ###############################################################################
+def _get_archive_object(case):
+###############################################################################
+    """
+    Returns the EnvArchive object associated with this case
+    """
+    caseroot = case.get_value("CASEROOT")
```

Comment:
The archive object already exists in a case and you don't need to instantiate a new one
archive = case.get_env("archive")